### PR TITLE
fix(pi07): gate optional prefix tokens in low- and high-level planners

### PR DESCRIPTION
**Pull request by shuheng-liu**
## What this does

Backports the #226-equivalent prefix-gating fixes from `pi07_paligemma` (commit `b7d3a82` on `claude/fix-issue-226-gt8z3`, merging into `fix/pi07_paligemma_fixes` then into `main`) to `pi07`'s low- and high-level planners.

**Low-level planner** (`pi07/low_level_planner/modeling_pi07_low_level.py::embed_prefix`) had three remaining divergences when all optional middle blocks (response / subgoal / metadata) are dropped:

- State-end was unconditionally `", "` (2 tokens) instead of `":\n"`.
- `";\n "` prefix-end was unconditionally appended even with no optional content to terminate, dangling 2 spurious tokens after the state-end.
- `"Action: "` indicator att_masks were `[1] + [0]*(N-1)` (single bidirectional block); `pi05` uses `[1]*N` (one causal block per token), which shifts the cumsum at the indicator → first-discrete boundary and changes the discrete-action CE loss.

The fix computes `has_any_optional` once at the top of `embed_prefix` and gates the state-end string + the prefix-end block on it. Indicator att_masks switched to `[1]*N` to match `pi05`. (The earlier `2949e73` commit only ported the response/subgoal/metadata content gating, not the surrounding separators or indicator att_masks.)

**High-level planner** (`pi07/high_level_planner/modeling_pi07_high_level.py::embed_prefix`) had one of the three issues — `";\n "` was emitted unconditionally, but it is the `metadata → "Updated Memory:"` separator. With no metadata there's nothing to terminate. Wrapped under `if metadata_tokens is not None:`. The `"Updated Memory: "` AR anchor itself stays unconditional because inference relies on it (memory_tokens is None at inference by design).

A byte-identity test against `pi06` (analogous to the `pi07_paligemma` vs `pi05` test in #226) is deliberately deferred to a follow-up issue (#230): `pi07_low_level` uses continuous state with separate `State:`/`Action:` indicators while `pi06` inlines discretized state in the language prompt, and `pi07_high_level` uses a different prompt template (`"Task: {task}, Past Memory: {past_mem}, State: {state}, "` vs `pi06`'s `"Task: {task}<eos>State: {state}<eos>Actions:"`), so byte-equivalence isn't feasible without aligning `pi06`'s prompts (or building `pi06_mem`).

### Drive-by fixes pulled from `main` to unblock CI

- `feat(ci): pin Claude workflows to Opus 4.7 1M, allow test-skip on doc fixes (#208)` — workflow file must match `main` to satisfy GitHub workflow validation; was failing the `Claude PR Review` check.
- `fix(ci): gate extract-claude-lessons on head branch, not PR author (#212)` — same reason.
- `ci: skip pi07_paligemma low-level test (broken at import) to unblock CPU CI (#209)` — `feat/pi07` was missing the `--ignore=tests/policies/test_pi07_paligemma_low_level_planner.py` flag in `cpu_test.yml`; the test file fails at import (`VJEPA2VideoEncoder` removed in `pi05_mem`) and that crashed CPU CI.
- `ci: ignore broken pi07 test in GPU CI; dedupe TODO in cpu_test (#217)` — same flag missing in `gpu_test.yml` + dedup.
- `fix(test): override actions stats shape in pi06 GPU smoke (PR #214)` — `test_complete_pi06_pipeline_integration_smoke` was failing on real GPU because the fixture stats shape `(50, 32)` didn't match `chunk_size=10`. Pre-existing latent bug; surfaced when running the GPU suite end-to-end on `mlbox`.

### Test fix surfaced by the GPU run

- `_verify_position_ids` and `_verify_action_expert_attention_mask` in `test_pi07_low_level_planner.py` only excluded `DISCRETE_ACTION_MAX_LENGTH` from the cross-attention prefix slice. The model (post-`2949e73`) already excludes the `"Action: "` indicator length (3 tokens via Gemma 3 tokenizer) too, matching `pi05`'s `discrete_action_indicator_max_length` logic — so the action expert sees the same prefix length at train and inference. The 3-token gap fails the `suffix_position_ids` assertion as soon as the GPU integration test runs. Latent break since `2949e73`; never caught because the test is gated by `@pytest.mark.gpu`.

## How it was tested

- `pre-commit run --all-files` — all hooks pass.
- **CPU subset** (`pytest -m "not gpu" -n auto tests/policies/`): 125 passed, 2 skipped, 1 pre-existing collection error in `test_pi07_paligemma_low_level_planner.py` (now excluded from CI by the `--ignore` flag pulled from #209/#217).
- **Full GPU suite** on `mlbox` (RTX 5090, 32 GB): `pytest -m gpu -n 0 -v --ignore=tests/policies/test_pi07_paligemma_low_level_planner.py tests/` → **20 passed, 2 skipped, 786 deselected, 0 failures, no OOM, 6m10s.** Includes `tests/policies/test_pi07_low_level_planner.py::test_complete_pi07_low_level_pipeline`, `test_pi07_high_level_planner.py::test_complete_pi07_high_level_planner_pipeline`, the regression class, and pi05 / pi05_mem / pi06 / pi0 / value smoke tests.
- Specifically `tests/policies/test_pi07_cpu.py::TestEmbedPrefixConditionalGuards` exercises the gating: `test_all_optional_blocks_absent_skips_emission` was updated to match the new layout (11 tokens, `":\n"` state-end, no trailing prefix-end); `test_mixed_subgoal_availability_zeros_pad_only_samples` and `test_response_mask_any_true_emits_block` still pass because they populate at least one optional, so `has_any_optional == True` and the layout matches the old expectations.
- Nightly regression tests (`regression_test.yml`) are deferred to the existing schedule.

## How to checkout & try? (for the reviewer)

```bash
git fetch origin
git checkout claude/fix-pi07-prefix-gating
uv sync --extra dev
pytest -m "not gpu" -n auto tests/policies/test_pi07_cpu.py tests/policies/test_pi07_low_level_planner.py tests/policies/test_pi07_high_level_planner.py
```

GPU subset:

```bash
pytest -m gpu -n 0 -v --ignore=tests/policies/test_pi07_paligemma_low_level_planner.py tests/
```

To re-verify the diff against the canonical `pi07_paligemma` fix:

```bash
git diff origin/feat/pi07..claude/fix-pi07-prefix-gating -- src/opentau/policies/pi07/low_level_planner/modeling_pi07_low_level.py
git show b7d3a82 -- src/opentau/policies/pi07_paligemma/low_level_planner/modeling_pi07_low_level.py
```

## Checklist

- [x] I have added Google-style docstrings to important functions and ensured function parameters are typed.
- [x] My PR includes policy-related changes.
  - [x] If the above is checked: I have run the GPU pytests (pytest -m "gpu") and regression tests.

GPU pytests run on `mlbox` (RTX 5090) — see "How it was tested" above. Nightly regression tests are deferred to the existing CI schedule. A follow-up issue (#230) is filed for `pi06` prompt alignment + future byte-identity tests against `pi07` planners.

### Note: Before submitting this PR, please read the [contributor guideline](https://github.com/TensorAuto/OpenTau/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: .github/workflows/claude-implement-fixes.yml [hunk 1]
```diff
@@ -100,6 +100,7 @@ jobs:
         with:
           anthropic_api_key: ${{ secrets.ANTHROPIC_API_KEY }}
           claude_args: |
+            --model ${{ vars.CLAUDE_MODEL || 'claude-opus-4-7[1m]' }}
```
[the remark this answers is not in the document]
Comment (shuheng-liu) [reply .github/workflows/claude-implement-fixes.yml:103]: The cherry pick is needed to trigger claude review

**Comment by shuheng-liu** [comment]
@claude fix per review, except for the `.github/workflows/claude-implement-fixes.yml` comment
